### PR TITLE
[usage] Add configmap for usage component to control run frequency

### PR DESCRIPTION
**Pull request by andrew-farries**
## Description

Make the frequency at which the usage component runs configurable via a configmap.

## Related Issue(s)
Fixes https://github.com/gitpod-io/gitpod/issues/10720
Part of #9036 

## How to test

* Open the preview environment.
* Edit the `usage-configmap` (`kubectl edit cm usage-config`) and change the `schedule` to `1m`.
* Kill the usage pod to restart it with the new config (`kubectl delete pod <usage pod>`).
* Tail the logs of the new usage pod, including timestamps (`kubectl logs -f --timestamps=true <usage-pod>`)
* Watch the logs of the usage component and see that it now runs at 1 minute intervals rather than the previous 1 hour.

## Release Notes

```release-note
NONE
```
* [x] /werft run with-payment=true

### CODE REVIEW

File: install/installer/pkg/components/usage/deployment.go
Comment (easyCZ) [inline install/installer/pkg/components/usage/deployment.go:82]: My 2 cents, we should move this into a config file just like the public API. I'm happy to approve here to unblock but longer term we're better off with most config in files.

**Review by easyCZ — APPROVED** [review]
/hold

In case you want to revisit how the config is provided (file instead of flag)

File: install/installer/pkg/components/usage/deployment.go
Comment (andrew-farries) [reply install/installer/pkg/components/usage/deployment.go:82]: You mean not have a `--schedule` flag at all but rather read the schedule from a config file?

I've tended to configure CLIs like this with multiple command line args, set via helm charts (or the installer here in the Gitpod case) in the past. What makes you prefer config files for this (not saying I disagree, just curious).

Comment (easyCZ) [reply install/installer/pkg/components/usage/deployment.go:82]: Generally the pattern used for other components. I think it also somewhat simplifies configuration for the installer and in general observing what a component is configured with (but that's debatable).

Comment (easyCZ) [reply install/installer/pkg/components/usage/deployment.go:82]: I think it's worth landing this is as is and align in subsequent PR.

Comment (andrew-farries) [reply install/installer/pkg/components/usage/deployment.go:82]: Made https://github.com/gitpod-io/gitpod/issues/10753 to track this.

**Review by corneliusludmann — APPROVED** [review]

**Pull request merged**
